apps/remote_control/lib/lexical/remote_control/compilation/dispatch.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Lexical.RemoteControl.Compilation.Dispatch do
+  alias Lexical.RemoteControl
+  alias Lexical.RemoteControl.Api.Messages
+  alias Lexical.RemoteControl.ModuleMappings
+  alias Lexical.RemoteControl.Plugin
+
+  import Messages
+  use GenServer
+
+  def dispatch(module_updated() = message) do
+    GenServer.cast(__MODULE__, {:dispatch, message})
+  end
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  def init(_) do
+    {:ok, nil}
+  end
+
+  def handle_cast({:dispatch, module_updated() = message}, state) do
+    module_updated(name: module_name, file: filename) = message
+    ModuleMappings.update(module_name, filename)
+    RemoteControl.notify_listener(message)
+    Plugin.on_module_updated(module_name)
+    maybe_report_progress(filename)
+    {:noreply, state}
+  end
+
+  defp maybe_report_progress(file) do
+    if Path.extname(file) == ".ex" do
+      file
+      |> progress_message()
+      |> RemoteControl.notify_listener()
+    end
+  end
+
+  defp progress_message(file) do
+    relative_path = Path.basename(file)
+    message = "compiling: " <> relative_path
```

Comment:
Two colons make the UI of VScode look ugly. Besides, I prefer `dirname/basename` here because it can provide more accurate information and is also shorter.

<img width="307" alt="image" src="https://github.com/lexical-lsp/lexical/assets/12830256/6f00bdaf-f3e1-404a-b8e8-e73836c24bf3">

```elixir
    relative_path = Path.relative_to_cwd(file)
    basename = Path.basename(file)
    message = file |> Path.dirname() |> Path.basename() |> Path.join(basename)
```
<img width="415" alt="image" src="https://github.com/lexical-lsp/lexical/assets/12830256/ad48d36f-72ba-4844-97ee-4e9e6b026043">

Reply:
That's interesting; my UI is completely different; I don't see the `mix compile` part at all

These messages go by so fast that they're not really viewable by mere mortals. The whole path was way too long, and my UI barely has space for the filename.

Reply:
The more I think about this, the more I think the message should just say "xxx files compiled"

Reply:
Right, Calculating percentages can bring a better UI, but it's difficult.

Reply:
It’s not really percentages, just a count of compiled files.